dev-infrastructure/templates/dev-acr.bicep

```diff
@@ -70,3 +93,91 @@ resource acrPurgeTask 'Microsoft.ContainerRegistry/registries/tasks@2019-04-01'
 
 @description('Login server property for later use')
 output loginServer string = acrResource.properties.loginServer
+
+resource pullCredential 'Microsoft.ContainerRegistry/registries/credentialSets@2023-01-01-preview' = if (length(quayRepositoriesToCache) > 0) {
+  name: 'quayPullCredential'
+  parent: acrResource
+  identity: {
+    type: 'SystemAssigned'
+  }
+  properties: {
+    authCredentials: [
+      {
+        name: 'Credential1'
+        passwordSecretIdentifier: 'https://${globalKeyVaultName}${env.suffixes.keyvaultDns}/secrets/${passwordSecretIdentifier}'
```

Comment:
since you lookup the keyvault with `existing` you can leverage `globalKeyVault.properties.vaultUri` for the KV base URL